src/shared/event-emitter.ts

```diff
@@ -0,0 +1,49 @@
+export type EventMap = Record<string, (...args: any) => void>;
```

Comment:
Perhaps it's better to define the event handler as `(...args: unknown[]) => unknown`?

Reply:
That results in type checking errors like this:

```
src/sidebar/services/toast-messenger.ts:47:57 - error TS2344: Type 'Events' does not satisfy the constraint 'EventMap'.
  Property 'toastMessageAdded' is incompatible with index signature.
    Type '(msg: ToastMessage) => void' is not assignable to type '(...args: unknown[]) => void'.
      Types of parameters 'msg' and 'args' are incompatible.
        Type 'unknown' is not assignable to type 'ToastMessage'.

47 export class ToastMessengerService extends EventEmitter<Events> {
```

So the more liberal `any` is still needed in this context.